protocols/imap/src/main/java/org/apache/james/imap/encode/main/DefaultImapEncoderFactory.java

```diff
@@ -48,10 +57,36 @@
 import org.apache.james.imap.encode.XListResponseEncoder;
 import org.apache.james.imap.encode.base.EndImapEncoder;
 
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableList;
+
 /**
  * TODO: perhaps a POJO would be better
  */
 public class DefaultImapEncoderFactory implements ImapEncoderFactory {
+    private static class DefaultImapEncoder implements ImapEncoder {
```

Comment:
is the ImapEncoder interface still useful?

Reply:
Yes as I need one item to be writing *any* kind of message and not just a single 'sub-class'.

(so do 'unknownResponseEncoder): I'm not able to implement acceptableMessages for them, thus this do not fit the ImapResponseEncoder hence the ImapEncoder separate interface...